### PR TITLE
The Onboarding flow revision

### PR DESCRIPTION
**Pull request by saledjenic**
The following flows are updated within this PR (since there were some design changes since they were initially developed), they should match the figma now:
- onboarding flow `I'm new to Status` -> `Generate keys for a new Keycard`
- onboarding flow `I'm new to Status` -> `Import a seed phrase` -> `Import a seed phrase into a new Keycard`
- onboarding flow `I already use Status` -> `Login with Keycard`
- login flow

Fixes: #7629

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: src/app/modules/startup/internal/keycard_create_pin_state.nim [hunk 1]
```diff
@@ -13,11 +13,15 @@ proc delete*(self: KeycardCreatePinState) =
 method executeBackCommand*(self: KeycardCreatePinState, controller: Controller) =
   controller.setPin("")
   controller.setPinMatch(false)
+  if self.flowType == FlowType.FirstRunNewUserNewKeycardKeys or
```
Comment (MishkaRogachev) [inline src/app/modules/startup/internal/keycard_create_pin_state.nim:16]: this check appears several times. Can it be generalised with static list: if KEYCARD_FLOW_TYPES.find(self.flowType)

**Review by borismelnik — APPROVED** [review]
LGTM :)

File: src/app/modules/startup/internal/state_factory.nim [hunk 2]
```diff
@@ -85,8 +97,12 @@ proc createState*(stateToBeCreated: StateType, flowType: FlowType, backState: St
     return newKeycardPluginReaderState(flowType, backState)
   if stateToBeCreated == StateType.KeycardInsertKeycard:
```
Comment (MishkaRogachev) [inline src/app/modules/startup/internal/state_factory.nim:98]: This ifs sequence can be a bit simplified with mapping <flowType, method>

**Review by osmaczko — APPROVED** [review]

**Review by MishkaRogachev — APPROVED** [review]
It's really hard to review such big PRs

**Review by endulab — APPROVED** [review]

**Comment by elina2015** [comment]
image

**Review by elina2015 — APPROVED** [review]
Tested and approved

**Pull request merged**